docs: add dynamic routes page (#219)
tuono-labs/tuono: apps/documentation/src/routes/documentation/routing/dynamic-routes.mdx

> 🚧 Only available for SSR apps. This feature will be soon available also for static generated apps.

When the exact segment names are not known in advance and you need to generate routes from dynamic data,
you can use dynamic segments, which are populated at request time.

## Convention

A dynamic segment can be created by wrapping a
file or a folder name in square brackets: `[segmentName]`.

> You are going to probably use something like:

Author: I'm not a fan of "personalizing" the documentation by pointing directly the reader. 
I'll think about a rephrase over this sentence

A dynamic segment can be created by wrapping a
file or a folder name in square brackets: `[segmentName]`.

> The dynamic segment can be used in both folder and file name:

Reviewer: ```suggestion
The dynamic segment can be used in folders and files as well:
```

- `src/routes/blog/[slug].tsx`
- `src/routes/blog/[slug]/index.tsx`

> The same path can have multiple dynamic segments:

Reviewer: ```suggestion
A path can contain multiple dynamic segments, allowing for more flexible routing.
```
